File: IT/1/Variabili, operatori, strutture di controllo/Esercizi/Python/esercizio31.py
# file esercizio31.py
# -*- coding: utf-8 -*-

from __future__ import print_function, division

if hasattr(__builtins__, 'raw_input'): input = raw_input

l = input("Quanti numeri vuoi inserire? ")

if not l.isdigit():
	print("Numero non valido!")
	exit()

limit = int(l)

i = 0
tot = 0

while i < limit:
	s = input("Numero corrente: ")

	if not s.replace("-", "").isdigit():
		print("Numero non valido!")
		continue

	n = int(s)

	tot += n

	i += 1

print("La media di tutti i numeri è {}.".format(tot / limit))

File: IT/1/Variabili, operatori, strutture di controllo/Esercizi/Java/Esercizio31.java
/* file Esercizio31.java */

import java.io.Console;

public class Esercizio31 {
	public static void main(String [] args) {
		Console console = System.console();

		String l = console.readLine("Quanti numeri vuoi inserire? ");

		if (!l.matches("[\\d]+")) {
				System.out.println("Numero non valido!");
				return;
		}

		int limit = Integer.parseInt(l);

		int i = 0;
		int tot = 0;

		while (i < limit) {
			String s = console.readLine("Numero corrente: ");

			if (!s.matches("[\\d]+")) {
				System.out.println("Numero non valido!");
				continue;
			}

			int n = Integer.parseInt(s);

			tot += n;
			i++;
		}

		System.out.println("La media di tutti i numeri è " + (double) tot / limit + ".");
	}
}

